File: backend/src/main/java/com/gollner/processflow/services/ClientService.java
package com.gollner.processflow.services;

import com.gollner.processflow.dto.clients.request.ClientRequestDTO;
import com.gollner.processflow.dto.clients.response.ClientMinDTO;
import com.gollner.processflow.entities.Client;
import com.gollner.processflow.entities.User;
import com.gollner.processflow.repositories.ClientRepository;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.UUID;

@Service
public class ClientService {

    private final ClientRepository repository;
    private AuthService authService;

    public ClientService(ClientRepository repository, AuthService authService) {
        this.repository = repository;
        this.authService = authService;
    }

    @Transactional(readOnly = true)
    public Page<ClientMinDTO> findAllByName(Pageable pageable, String name) {
        User responsibleUser = authService.getAuthenticatedUser();

        Page<Client> clients = repository.findAllByName(pageable, name, responsibleUser.getId());
        return clients.map(ClientMinDTO::new);
    }

    @Transactional
    public ClientMinDTO insert(ClientRequestDTO clientRequestDTO) {
        User responsibleUser = authService.getAuthenticatedUser();

        Client client = new Client();
        copyDtoToEntity(client, clientRequestDTO);
        client.setResponsibleUser(responsibleUser);

        return new ClientMinDTO(repository.save(client));
    }

    public void update(UUID id, ClientRequestDTO clientRequestDTO) {
        Client client = repository.findById(id).orElseThrow(() -> new RuntimeException("Cliente não encontrado"));

        authService.verifyResponsible(client);

        copyDtoToEntity(client, clientRequestDTO);

        repository.save(client);
    }

    @Transactional
    public void delete(UUID id) {
        Client client = repository.findById(id).orElseThrow(() -> new RuntimeException("Cliente não encontrado"));

        authService.verifyResponsible(client);

        repository.deleteById(id);
    }

    private void copyDtoToEntity(Client client, ClientRequestDTO clientRequestDTO) {
        client.setDocument(clientRequestDTO.document());
        client.setEmail(clientRequestDTO.email());
        client.setName(clientRequestDTO.name());
        client.setPhone(clientRequestDTO.phone());
    }
}

File: frontend/src/services/client-service.ts
import { AxiosRequestConfig } from "axios";
import { ClientMinDTO, ClientRequestDTO } from "../models/client";
import { Paged } from "../models/generics";
import { requestBackend } from "../utils/request";

export async function findAllPaged(
    page: number,
    name: string,
    size = 8
): Promise<Paged<ClientMinDTO[]>> {
    const config: AxiosRequestConfig = {
        url: "/clients",
        method: "GET",
        params: { page, name, size }
    }
    console.log((await requestBackend(config)).data);
    return (await requestBackend(config)).data;
}

export async function insert(
    data: ClientRequestDTO
): Promise<ClientMinDTO> {
    const config: AxiosRequestConfig = {
        url: "/clients",
        method: "POST",
        data: data
    }

    return (await requestBackend(config)).data;
}

export async function deleteClient(id: string): Promise<void> {
    const config: AxiosRequestConfig = {
        url: `/clients/${id}`,
        method: "DELETE"
    }

    await requestBackend(config)
}

export async function update(id: string, client: ClientRequestDTO): Promise<void> {
    const config: AxiosRequestConfig = {
        url: `/clients/${id}`,
        method: "PUT",
        data: client
    }

    await requestBackend(config)
}